components/scim/org.wso2.carbon.identity.scim.common/src/main/java/org/wso2/carbon/identity/scim/common/utils/SCIMCommonUtils.java

```diff
@@ -158,10 +162,18 @@ public static String getPrimaryFreeGroupName(String groupName) {
         if (index > 0) {
             // Using the short-circuit. User name comes with the domain name.
             String domain = groupName.substring(0, index);
-            if (UserCoreConstants.PRIMARY_DEFAULT_DOMAIN_NAME.equals(domain)) {
+            if (StringUtils.equals(UserCoreConstants.PRIMARY_DEFAULT_DOMAIN_NAME, domain)) {
                 return groupName.substring(index + 1);
```

Comment:
As I see it's not needed to use StringUtils here. The variable is compared with a String constant which will not be null